services/crawler/src/providers/Iotex.ts

```diff
@@ -0,0 +1,260 @@
+              date: new Date(action.timestamp.seconds * 1000).toISOString(),
+              staked_candidate: core?.stakeCreate?.candidateName,
+              staked_amount: core.stakeCreate?.stakedAmount,
+              staked_duration: core.stakeCreate?.stakedDuration,
+              auto_stake: core.stakeCreate?.autoStake,
+          }
+      case "stakeAddDeposit":
+          return {
+            type: IotexActionType.stakeAddDeposit,
+            date: new Date(action.timestamp.seconds * 1000).toISOString(),
+            amount: core.stakeAddDeposit?.amount,
+            payload: core.stakeAddDeposit?.payload !== undefined ? Buffer.from(core.stakeAddDeposit?.payload).toString('hex') : '',
+            bucket_index: core.stakeAddDeposit?.bucketIndex
+          }
+      case "execution":
+          return {
+            type: IotexActionType.execution,
+            date: new Date(action.timestamp.seconds * 1000).toISOString(),
+            contract: core.execution?.contract,
+            amount: core.execution?.amount,
+            data: core.execution?.data !== undefined ? Buffer.from(core.execution.data).toString("hex") : '',
+          }
+          case "putPollResult":
+          return {
+            type: IotexActionType.putPollResult,
+            date: new Date(action.timestamp.seconds * 1000).toISOString(),
+            candidates: core.putPollResult?.candidates?.candidates.map(c => c.address) ?? [],
+          }
+      case "stakeWithdraw":
+          return {
+            type: IotexActionType.stakeWithdraw,
+            date: new Date(action.timestamp.seconds * 1000).toISOString(),
+            bucket_index: core.stakeWithdraw?.bucketIndex,
+            payload: core.stakeWithdraw?.payload !== undefined ? Buffer.from(core.stakeWithdraw.payload).toString('hex') : '',
+          }
+      case "stakeChangeCandidate":
+          return {
+            type: IotexActionType.StakeChangeCandidate,
+            date: new Date(action.timestamp.seconds * 1000).toISOString(),
+            candidate: core.stakeChangeCandidate?.candidateName,
+            bucket_index: core.stakeChangeCandidate?.bucketIndex,
+            payload: core.stakeChangeCandidate?.payload !== undefined ? Buffer.from(core.stakeChangeCandidate.payload).toString('hex') : '',
+          }
+      case "stakeRestake":
+          return {
+            type: IotexActionType.stakeRestake,
+            date: new Date(action.timestamp.seconds * 1000).toISOString(),
+            payload: core.stakeRestake?.payload !== undefined ? Buffer.from(core.stakeRestake.payload).toString('hex') : '',
+            auto_stake: core.stakeRestake?.autoStake,
+            staked_duration:core.stakeRestake?.stakedDuration,
+            bucket_index: core.stakeRestake?.bucketIndex
+          }
+          default:
+            console.log(type)
+    }
+  }
+}
+
+export async function newIotexService (opt?: IotexServiceOptions): Promise<IotexService> {
```

Comment:
Returns `IotexService`

Reply:
I wanted to avoid name collisions if it ever happens with other providers. I wanted to add a prefix like `T{className}` or in the case of provider`{className}Provider`

Reply:
Just pointing out the promise! Looks leftover.